go/vt/vttablet/queryservice/wrapped.go

```diff
@@ -84,8 +84,20 @@ func wrapInVT15001(err error, inTx bool) error {
 	}
 	c := vterrors.Code(err)
 	m := err.Error()
-	if c == vtrpcpb.Code_UNAVAILABLE && strings.Contains(m, "connection refused") {
-		return vterrors.VT15001(c, m)
+
+	conditions := []struct {
+		code        vtrpcpb.Code
+		containsMsg string
+	}{
+		{code: vtrpcpb.Code_UNAVAILABLE, containsMsg: vterrors.ConnectionRefused},
+		{code: vtrpcpb.Code_FAILED_PRECONDITION, containsMsg: vterrors.WrongTablet},
+		{code: vtrpcpb.Code_CLUSTER_EVENT, containsMsg: vterrors.NotServing},
+		{code: vtrpcpb.Code_CLUSTER_EVENT, containsMsg: vterrors.ShuttingDown},
+	}
+	for _, condition := range conditions {
+		if c == condition.code && strings.Contains(m, condition.containsMsg) {
+			return vterrors.VT15001(c, m)
+		}
 	}
```

Comment:
better to have a map over slice for conditions.
It should be created once outside of `wrapInVT15001` method

Reply:
I think you were seeing an old version of the code, see https://github.com/vitessio/vitess/pull/17669/files#diff-51ee5faf2cef1190577f7bfe3d40463453b58f8fde1b90800c2967057c5b074a for the latest version that use a `swtich`